tests/extension.spec.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import assert from 'assert';
+import { test, expect } from './fixtures.js';
+
+test.skip(({ mcpExtension }) => !mcpExtension);
+
+test('test reopen browser', async ({ client, clientOutputLines, mcpExtensionPage }) => {
+  assert(mcpExtensionPage, 'mcpExtensionPage is required for this test');
+  await client.callTool({
+    name: 'browser_navigate',
+    arguments: {
+      url: 'data:text/html,<html><title>Title</title><body>Hello, world!</body></html>',
+    },
+  });
+
+  expect(await client.callTool({
+    name: 'browser_close',
+    arguments: {},
+  })).toContainTextContent('No open pages available');
+
+  expect(await client.callTool({
+    name: 'browser_navigate',
+    arguments: {
+      url: 'data:text/html,<html><title>Title</title><body>Hello, world!</body></html>',
+    },
+  })).toContainTextContent('Error: browser_connect is required before launching the browser');
+
+  const browserContextCall = client.callTool({
+    name: 'browser_connect',
+    arguments: {},
+  });
+  await expect.poll(() => clientOutputLines.filter(line => line.startsWith('open call to: ')), { timeout: test.info().timeout }).toHaveLength(2);
+  const openCallURL = clientOutputLines.filter(line => line.startsWith('open call to: '))[1].split('open call to: ')[1];
+  await mcpExtensionPage.goto(openCallURL);
+  await mcpExtensionPage.getByRole('button', { name: 'Allow Connection' }).click();
+  await expect(mcpExtensionPage.getByRole('heading', { name: 'Connection Established' })).toBeVisible();
+  await browserContextCall;
+
+  expect(await client.callTool({
```

Comment:
let's double-check that `mcpExtensionPage` is also showing that content now, so we can be sure it's using the right browser